sympy/printing/latex.py

```diff
@@ -767,10 +768,21 @@ def _print_UndefinedFunction(self, expr):
         return self._hprint_Function(str(expr))
 
     def _print_FunctionClass(self, expr):
-        if hasattr(expr, '_latex_no_arg'):
-            return expr._latex_no_arg(self)
-
-        return self._hprint_Function(str(expr))
+        from sympy.functions.special.tensor_functions import KroneckerDelta
+        from sympy.functions.special.gamma_functions import gamma, lowergamma
+        from sympy.functions.special.beta_functions import beta
+        from sympy.functions.special.delta_functions import DiracDelta
+        from sympy.functions.special.error_functions import Chi
+        special_cases = {KroneckerDelta: r'\delta',
+                         gamma:  r'\Gamma',
+                         lowergamma: r'\gamma',
+                         beta: r'\operatorname{B}',
+                         DiracDelta: r'\delta',
+                         Chi: r'\operatorname{Chi}'}
```

Comment:
I think these should be centralized. Could you move them to the top and use them in the respective function printing methods, or use `_print_FunctionClass` for printing function's name in their printing methods?

Reply:
Or I think we can move special functions handling except `KroneckerDelta` and `DiracDelta` to `_print_Function`, as their methods are identical and probably other special functions we might add will be same too.

Reply:
I like the idea of centralizing things even further, i.e., removing the various `_print_gamma`, `_print_lowergamma` etc and just adding a dictionary of special cases to `_print_Function`.  Most of them don't do anything special apart from custom name. But this feels like a separate project to me; this PR already touches many things at once.

Reply:
I think @ylemkimon just means to move this dictionary up to the class level, so it is easier for subclasses to override (otherwise they'd have to redefine this entire function).

Reply:
Circular imports seem to prevent me from doing so.  Hopefully not much of an issue, because this function does not have much besides those imports and that dictionary.

Reply:
You can work around that by making it a @property method. It won't be a class attribute in that case, but I don't think that's a big deal.